src/codec.rs

```diff
@@ -488,29 +550,122 @@ impl<T, X> Encode for X where
 pub trait WrapperTypeDecode: Sized {
```

Comment:
Is it still marker trait?

Reply:
From the user's perspective, yes; from the implementation point of view, no. Practically it doesn't make much difference, I think.